components/record-page.tsx

```diff
@@ -43,30 +54,36 @@ const ActionButtons = ({
   </div>
 );
 
-const getAudioContext = () => window.AudioContext || window.webkitAudioContext;
+const getAudioContext = () => typeof window !== undefined && window.AudioContext || window.webkitAudioContext;
+
+type DeviceItems = MediaDeviceInfo[];
+type DeviceList = {
+  video: DeviceItems;
+  audio: DeviceItems;
+};
 
-function RecordPage () {
-  const [file, setFile] = useState(null);
-  const [startRecordTime, setStartRecordTime] = useState(null);
+const RecordPage: React.FC<Record<never, never>> = () => {
```

Comment:
This is an Uneducated Question, but is there a reason that each of these are separate state hooks rather than a combined object? Performance concern?

Reply:
Hooks convention is to separate each state value in it's own `useState`

if you do `useState` with an object then it gets a little messy, where you have to spread the current state every time you want to change a single value:

```
 const [state, setState] = useState({ a: 1, b: 2})

setState({...state, b: 3 })
```

Reply:
Yeah, I know--I come from more of a functional-ish backgrounds so the latter seems more readable to me, but I don't have a strong feeling there.